src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSBwaUtils.java

```diff
@@ -0,0 +1,66 @@
+package org.broadinstitute.hellbender.tools.spark.pathseq;
+
+import com.google.cloud.dataflow.sdk.options.PipelineOptions;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import org.apache.spark.api.java.JavaRDD;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceMultiSource;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.SerializableFunction;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.*;
+
+/**
+ * Utility functions for PathSeq Bwa tool
+ */
+public final class PSBwaUtils {
```

Comment:
It's essential. For some reason GATK is very slow at reading header sequences. This optimization cuts the final Score tool run time from ~5 min to ~30 seconds.